### PR TITLE
[production/RRFS.v1] Reorganization of NWGES directory structure and removal of SLASH_ENSMEM_SUBDIR

### PR DESCRIPTION
**Pull request by BenjaminBlake-NOAA**
## DESCRIPTION OF CHANGES: 

- When I started working on reorganizing the run directory structure for the remaining DA tasks in the RRFS workflow, I realized that two things also need to be modified: the NWGES directory structure, and the removal of SLASH_ENSMEM_SUBDIR.  This PR does that for the non-DA and firewx workflows to keep things simple.
- The NWGES directories are re-structured to be more similar to COMOUT, so the cycle-specific directories are replaced by ${RUN}.${PDY}/${cyc} directories.  The filenames in nwges will also contain ${PDY}${cyc} in them.  To see an example of this on Cactus, you can look in /lfs/h2/emc/ptmp/Benjamin.Blake/test_nonDA_netcdf/nwges.  In addition, `$GESROOT` is now used instead of ${NWGES} for the path to the nwges directory.
- The SLASH_ENSMEM_SUBDIR variable is removed.  If the task/directory is for an RRFS ensemble member, the member number will be incorporated into the directory name itself using the ${mem_num} variable.
- Also included in this PR are the changes from PR #353 which have not been added to production/RRFS.v1 yet (exrrfs_run_prdgen.sh changes).

## TESTS CONDUCTED: 

These changes were tested with the non-DA engineering test and the RRFS fire weather workflow.

### Machines/Platforms:

- WCOSS2
  - [x] Cactus/Dogwood
  - [ ] Acorn
- RDHPCS
  - [ ] Hera
  - [ ] Jet
  - [ ] Orion
  - [ ] Hercules

### Test cases: 

- [ ] Engineering tests
  - [x] Non-DA engineering test
  - [ ] DA engineering test
    - [ ] Retro
    - [ ] Ensemble
    - [ ] Parallel
- [x] RRFS fire weather
- [ ] RRFS_A:
- [ ] RRFS_B:
- [ ] RTMA:
- [ ] Others:

## ISSUE: 

- Continues work on issue #237

### CODE REVIEW

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA  Having a NWGES directory structure similar to COMOUT also makes sense to me if we want to go that route.  So for example we could have directories within nwges like this instead of the current ${PDY}${cyc} directories:
rrfs.${PDY}/${cyc}
enkfrrfs.${PDY}/${cyc}
refs.${PDY}/${cyc}

**Comment by MatthewPyle-NOAA** [comment]
@BenjaminBlake-NOAA I like that overall structure

**Comment by BenjaminBlake-NOAA** [comment]
@MatthewPyle-NOAA  Sounds good to me as well.  I've updated my PR to reflect that structure and tested the changes.

**Comment by MatthewPyle-NOAA** [comment]
FWIW, I do use a GESROOT definition in the HREF system.  I know I was surprised that NCO doesn't really talk about it in the production standards - need to dig out some old e-mail conversations with Steven Earle about it.  But in HREF I do stuff like:

export GESROOT=${GESROOT:-$(compath.py $envir/$NET/${href_ver})/nwges}

File: jobs/JRRFS_RUN_PRDGEN
Comment (MatthewPyle-NOAA) [inline jobs/JRRFS_RUN_PRDGEN:1]: One comment that applies to RUN_POST as well - I don't believe there are "spinup" cycles for the ensemble forecasts.

**Review by MatthewPyle-NOAA — COMMENTED** [review]
Generally looked good to me - but did have a comment about spinup cycles.

File: jobs/JRRFS_RUN_PRDGEN
Comment (BenjaminBlake-NOAA) [reply jobs/JRRFS_RUN_PRDGEN:1]: You're right, thanks for catching that.  I'll update those J-jobs.

**Comment by MatthewPyle-NOAA** [comment]
@BenjaminBlake-NOAA Did you retest with your latest changes in this PR?

**Comment by MatthewPyle-NOAA** [comment]
No need to run more tests at this point - will approve and get it merged.

**Review by MatthewPyle-NOAA — APPROVED** [review]
Looks good, but a complete test will require testing the full system.  So will merge now so as not to hold things up.

**Pull request merged**